experimenter/experimenter/experiments/constants.py

```diff
@@ -54,9 +59,38 @@ class ApplicationConfig:
     slug: str
     app_name: str
     channel_app_id: Dict[str, str]
-    kinto_collection: str
+    default_kinto_collection: str
     randomization_unit: str
     is_web: bool
+    kinto_collections_by_feature_id: Optional[dict[str, str]] = field(default=None)
+
+    def get_kinto_collection_for(self, experiment: NimbusExperiment) -> str:
+        # If the experiment in question contains any features in the feature to
+        # collection map, then *all* features must be present in the map and
+        # they all must point to the same collection.
+        if self.kinto_collections_by_feature_id is not None:
+            target_collections = {
+                self.kinto_collections_by_feature_id.get(
+                    feature.slug, self.default_kinto_collection
+                )
+                for feature in experiment.feature_configs.all()
+            }
+
+            if len(target_collections) == 1:
+                return next(iter(target_collections))
+
+            elif len(target_collections) > 1:
+                raise AssertionError("Experiment targets multiple collections")
```

Comment:
Ah yeah I didn't think of it here, this should be handled in the review serializer.